doctr/models/detection/core.py

```diff
@@ -68,12 +70,9 @@ def box_score(
             polygon objectness
         """
         h, w = pred.shape[:2]
-        xmin = np.clip(np.floor(points[:, 0].min()).astype(np.int), 0, w - 1)
-        xmax = np.clip(np.ceil(points[:, 0].max()).astype(np.int), 0, w - 1)
-        ymin = np.clip(np.floor(points[:, 1].min()).astype(np.int), 0, h - 1)
-        ymax = np.clip(np.ceil(points[:, 1].max()).astype(np.int), 0, h - 1)
-
-        return pred[ymin:ymax + 1, xmin:xmax + 1].mean()
+        mask = np.zeros((h, w), np.int32)
+        cv2.fillPoly(mask, [points.astype(np.int32)], 1.0)
+        return np.sum(pred * mask) / np.count_nonzero(pred * mask)
```

Comment:
Product being computed twice, let's compute this earlier on to avoid the duplicate computation